### PR TITLE
Updating dependencies

### PR DESCRIPTION
**Pull request by vinicius0026**
As previously stated in #172, this is the first step to provide support to hapi v17.

All dependencies have been upgraded to latest version, except for hapi, that was upgraded to v16.

Tests were fixed to adhere to new syntax as per lab@15, but no changes were made to the actual code.

Support for node < 8 was dropped, since `async await` syntax is used in tests.

This is not intended to be released, but serves as an intermediary step since this is my first contribution to hapi ecosystem.

All comments are welcome.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 language: node_js
 
 node_js:
-  - "4"
-  - "6"
+  - "8"
```
Comment (Marsup) [inline .travis.yml]: You're missing "9" and "node". See hapi for that one.

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,2125 @@
+{
```
Comment (Marsup) [inline package-lock.json]: We don't use package locks, just add it to the .gitignore.

File: package.json [hunk 3]
```diff
@@ -12,20 +12,20 @@
...
   "dependencies": {
-    "boom": "3.x.x",
-    "hoek": "4.x.x",
-    "joi": "8.x.x"
+    "boom": "7.x.x",
+    "hoek": "5.x.x",
+    "joi": "13.x.x"
   },
   "peerDependencies": {
     "hapi": ">=11.x.x"
```
Comment (Marsup) [inline package.json]: You'll probably want to bump that.

File: test/index.js [hunk 4]
```diff
@@ -15,167 +15,138 @@ const internals = {};
 
 // Test shortcuts
 
-const lab = exports.lab = Lab.script();
-const describe = lab.describe;
-const it = lab.it;
-const expect = Code.expect;
+const { describe, it } = exports.lab = Lab.script();
```
Comment (Marsup) [inline test/index.js]: You can also get the `expect` in this destructuring.

File: test/index.js [hunk 5]
```diff
@@ -15,167 +15,138 @@ const internals = {};
...
+            method: 'GET', path: '/login/{user}',
+            config: {
+                auth: { mode: 'try' },
+                handler: function (request, reply) {
 
-                        request.cookieAuth.set({ user: request.params.user });
-                        return reply(request.params.user);
-                    }
+                    request.cookieAuth.set({ user: request.params.user });
+                    return reply(request.params.user);
```
Comment (Marsup) [inline test/index.js]: `reply` is not a v17 thing anymore, I'm surprised it would work.

File: test/index.js [hunk 6]
```diff
@@ -15,167 +15,138 @@ const internals = {};
...
+        });
 
-            server.route({
-                method: 'GET', path: '/login/{user}',
-                config: {
-                    auth: { mode: 'try' },
-                    handler: function (request, reply) {
+        server.route({
+            method: 'GET', path: '/login/{user}',
+            config: {
```
Comment (Marsup) [inline test/index.js]: `config` is now `options` if I'm not mistaken.

**Comment by Marsup** [comment]
Sorry, conf reviews are the worst, I didn't see your 2 steps thing, you can disregard :p

File: test/index.js (hunk 5 above)
Comment (alexandrzim) [reply test/index.js]: He said it's not intended to be released

**Comment by vinicius0026** [comment]
@Marsup No problem! I'll ignore the package-lock file and fix travis conf anyway.

(as per the lock file, I just committed it because it exists in hapi repo - as npm-shrinkwrap.json file - but I'll remove it)

As soon as this lands I'll start working on the actual update to support hapi v17.

File: test/index.js (hunk 4 above)
Comment (vinicius0026) [reply test/index.js]: Wouldn't it then get `expect` from lab instead of getting it from `Code`?

**Comment by mrlannigan** [comment]
@vinicius0026 Just so happens I worked on this today. ;) Welcome all comments on this https://github.com/hapijs/hapi-auth-cookie/pull/174.

**Comment by vinicius0026** [comment]
@mrlannigan That's unfortunate that we both worked on the same thing at the same time. But since your PR is a complete update to hapi v17 (as mine is still a WIP), I'll gladly close this once yours lands.

**Comment by mrlannigan** [comment]
@vinicius0026 Yea, my apologies for not posting it somewhere that I was going to work on it.

**Pull request closed without merging**